File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.PriorityQueue;
class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int s:scoville) {
            pq.offer(s);
        }
        
        int cnt = 0;
        while (pq.peek()<K && pq.size()>1) {
            int s1 = pq.poll();
            int s2 = pq.poll();
            pq.offer(s1+s2*2);
            cnt++;
        }
        
        return pq.peek()<K ? -1 : cnt;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    heap = []
    for s in scoville:
        heapq.heappush(heap, s)
        
    cnt=0
    
    while heap[0]<K:
        cnt+=1
        s1 = heapq.heappop(heap)
        s2 = heapq.heappop(heap)
        heapq.heappush(heap, s1+s2*2)
        if len(heap)==1:
            break
            
    if heap[0]<K:
        return -1
    else:
        return cnt